src/AspNet.Security.OAuth.Keycloak/KeycloakAuthenticationOptions.cs

```diff
@@ -36,6 +37,48 @@ public KeycloakAuthenticationOptions()
             ClaimActions.MapJsonKey(ClaimTypes.Role, "roles");
         }
 
+        /// <summary>
+        /// Gets or sets the value for Keycloak client's access type
+        /// </summary>
+        public KeycloakAuthenticationAccessType AccessType { get; set; }
+
+        /// <inheritdoc />
+        public override void Validate()
+        {
+            try
+            {
+                // HACK
+                // We want all of the base validation except for ClientSecret,
+                // so rather than re-implement it all, catch the exception thrown
+                // for that being null and only throw if we aren't using public access type.
+                // This does mean that three checks have to be re-implemented
+                // because the won't be validated if the ClientSecret validation fails.
+                base.Validate();
+            }
+            catch (ArgumentException ex) when (ex.ParamName == nameof(ClientSecret))
+            {
+                if (AccessType != KeycloakAuthenticationAccessType.Public)
+                {
+                    throw;
+                }
+            }
```

Comment:
This could be simplified to just swallow in the required case, rather than need to throw when it isn't.

```suggestion
            catch (ArgumentException ex) when (ex.ParamName == nameof(ClientSecret) && AccessType == KeycloakAuthenticationAccessType.Public)
            {
                // No client secret is required for a public key.
                // See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/issues/610
            }
```